clients/src/main/java/org/apache/kafka/clients/NetworkClientUtils.java

```diff
@@ -76,6 +75,30 @@ public static boolean awaitReady(KafkaClient client, Node node, Time time, long
         return client.isReady(node, attemptStartTime);
     }
 
+    public static boolean awaitReady(KafkaClient client, Node node, Time time, long timeoutMs, Boolean fastTimeout) throws IOException {
```

Comment:
why use Boolean instead of primitive ?

Reply:
@tedyu Boolean 's type is reference?

Reply:
Seems the `fastTimeout` should be thread-safe since it will be accessed by both RequestSendThread and controller event thread. Perhaps `volatile` is a least requirement for memory visibility.

Reply:
I changed the type to boolean here (and on line 89) - compilation passed.

This would avoid unboxing.

Reply:
@huxihx @tedyu thanks for your reply. I modify the code:
1. ensure  requirement for memory visibility.
2. weakup up the poll.
3. fastTimeout  thread safe.